scripts-test/weather.py

```
@@ -0,0 +1,75 @@
+    try:
+        weather_data = get_weather(city)
+        print_weather(weather_data)
+    except requests.HTTPError as e:
+        print(f"HTTP Error occurred: {e}")
+    except requests.ConnectionError:
+        print("Connection Error: Please check your internet connection.")
+    except requests.Timeout:
+        print("Request timed out. The weather service might be slow or unavailable.")
+    except (ValueError, KeyError) as e:
+        print(f"Data Error: Could not process the weather information: {e}")
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)